src/lib/consensus/proof_of_stake.ml

```diff
@@ -2600,7 +2600,12 @@ module Hooks = struct
           : (Get_epoch_ledger.query, Get_epoch_ledger.response) rpc
 
     type rpc_handler =
-      | Rpc_handler : ('q, 'r) rpc * ('q, 'r) rpc_fn -> rpc_handler
+      | Rpc_handler :
```

Comment:
is there a way to avoid repeating this type definition?

Reply:
I spent a while trying and it was difficult to do without repeating a lot of code or changing some stuff around significantly.